# Principal Component Analysis

## Libraries

In [83]:
# Libraries to handle data
import numpy as np
import pandas as pd
from scripts import data_wrangling as dw

In [84]:
# Magic commands
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


## Datasets

Since PCA is restricted to numerical variables, we must filter categorical variables from the datasets to properly perform this method. The class `Dataset` from the Python script `scripts/data_wrangling.py` attends to this task.

In [90]:
# Dataset object from the script data_wrangling
dataset = dw.Dataset()

mat_df = dataset.get_df(subject = 'mat', variables = 'numerical')
por_df = dataset.get_df(subject = 'por', variables = 'numerical')

# Let us rescale the grades so that 
# the maximum will be 10 instead of 20
#---

mat_df['G1'] *= 0.5
mat_df['G2'] *= 0.5
mat_df['G3'] *= 0.5

por_df['G1'] *= 0.5
por_df['G2'] *= 0.5
por_df['G3'] *= 0.5

## Mathematics

In [98]:
mat_df.describe().transpose()

count       mean       std   min   25%   50%   75%   max
age       395.0  16.696203  1.276043  15.0  16.0  17.0  18.0  22.0
failures  395.0   0.334177  0.743651   0.0   0.0   0.0   0.0   3.0
absences  395.0   5.708861  8.003096   0.0   0.0   4.0   8.0  75.0
G1        395.0   5.454430  1.659597   1.5   4.0   5.5   6.5   9.5
G2        395.0   5.356962  1.880752   0.0   4.5   5.5   6.5   9.5
G3        395.0   5.207595  2.290721   0.0   4.0   5.5   7.0  10.0

From the dataframe above:

- the majority of students are $17 \pm 1$ years old since the first quartile of the age variable is 16 and the third quartile is 18;

- Most students do not have past class failures since the third quartile of the failures variable is 0;

- Most students are present in most classes;

- The first, second, and final grades are similar centered in 5.5.